## Simple Linear Regresion 

### Importing the Libraries 

In [5]:
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt 
%matplotlib inline

### Importing The Dataset 

In [9]:
df = pd.read_csv('Salary_Data.csv')

### Displaying the Dataset 

In [10]:
df.head()

,YearsExperience,Salary
0,1.1,39343.0
1,1.3,46205.0
2,1.5,37731.0
3,2.0,43525.0
4,2.2,39891.0


In [11]:
print(df.shape)

(30, 2)


### Using Iloc to Split 

In [19]:
print(df.shape)
X = df.iloc[:,:-1].values 
y = df.iloc[:,-1].values

(30, 2)


### Splitting the Dataset into Training and Test Sets 

In [23]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=0)

### Training the Simple Linear Regression model on the Training set & getting the final linear equation with the values of coefficients

In [24]:
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,y_train)
print(regressor.coef_, regressor.intercept_)

[9345.94244312] 26816.192244031183


From the Simple Linear Regression formula (y = b0 + b1(x)), Here, b0 = 26816 and b1 = 9345. So, b1 tells us that the salary of an employee increases by 9345.94 with +1 year of Experience.

### Predicting the Test set result

In [27]:
prediction = regressor.predict(X_test)

### Model Evaluation

In [32]:
from sklearn.metrics import mean_squared_error, r2_score
import numpy as np

mse = mean_squared_error(y_test, prediction)
rmse = np.sqrt(mse)
r2 = r2_score(y_test, prediction)

print("MSE:", mse)
print("RMSE:", rmse)
print("r2:", r2)

MSE: 21026037.329511296
RMSE: 4585.4157204675885
r2: 0.9749154407708353


### Making a single prediction (for example the salary of an employee with 12 years of experience)

In [33]:
print(regressor.predict([[12]]))

[138967.5015615]
